packages/react/src/auto/polaris/tableCells/PolarisAutoTableTextCell.tsx

```diff
@@ -1,13 +1,30 @@
-import { Text } from "@shopify/polaris";
-import React from "react";
+import { Tooltip } from "@shopify/polaris";
+import React, { useEffect, useRef, useState } from "react";
 
 export const PolarisAutoTableTextCell = (props: { value: any }) => {
   const { value } = props;
   const stringifiedValue = typeof value === "object" ? ("markdown" in value ? value.markdown : JSON.stringify(value)) : value;
 
-  return (
-    <Text as="span" truncate>
+  const [isOverflowed, setIsOverflow] = useState(false);
+  const textElementRef = useRef<HTMLSpanElement>(null);
+
+  useEffect(() => {
+    setIsOverflow(!!textElementRef.current && textElementRef.current.scrollWidth > textElementRef.current.clientWidth);
+  }, []);
+
+  const polarisTextWithRef = (
+    <span className="Polaris-Text--root Polaris-Text--block Polaris-Text--truncate" ref={textElementRef}>
       {stringifiedValue}
-    </Text>
+    </span>
+  );
+
+  if (!isOverflowed) {
+    return polarisTextWithRef;
+  }
+
+  return (
+    <Tooltip content={stringifiedValue} hoverDelay={500}>
```

Comment:
![CleanShot 2024-07-26 at 15 45 07@2x](https://github.com/user-attachments/assets/cd496df9-dda9-4d38-8ea4-a35d72b8c09d)

If the length gets really extreme, the tooltip can spill out of the window. This `content` prop can take in react components though which might make it possible to put in a scrollable. 

Although, it might be easier to do that with a popover since it seems that Tooltips seem to flash in/out of existence upon mouse scrolling. Using `active={true}` in the tooltip wont force it to stay open either

Reply:
I spoke with Briggs about this -- I think the tooltip should be allowed to be wider than the column, and have its own vertical stroll within it

Reply:
@MillanWangGadget updated to widen the tooltip and scroll within the tooltip container